pkg/server/secrets-encrypt.go

```diff
@@ -186,6 +186,10 @@ func encryptionConfigHandler(ctx context.Context, server *config.Control) http.H
 			genErrorMessage(resp, http.StatusBadRequest, err)
 			return
 		}
+		// etcd will update very quickly, but there still some wait time before the files are written.
+		// If a user kills the k3s server immdiately after this call we run into issues were the files
+		// have not yet been written.
```

Comment:
codespell will probably nit on this block, but how about:
```suggestion
		// If a user kills the k3s server immediately after this call, we run into issues where the files
		// have not yet been written. This sleep ensures that things have time to sync to disk before
		// the request completes.
```